napari/_qt/dialogs/qt_package_installer.py

```diff
@@ -0,0 +1,419 @@
+            List of packages to uninstall.
+        prefix : Optional[str], optional
+            Optional prefix from which to uninstall packages.
+
+        Returns
+        -------
+        JobId : int
+            ID that can be used to cancel the process.
+        """
+        item = self._build_queue_item(
+            task_handler=task_handler,
+            action="uninstall",
+            pkgs=pkgs,
+            prefix=prefix,
+            **kwargs,
+        )
+        return self._queue_item(item)
+
+    def cancel(self, job_id: Optional[JobId] = None):
+        """Cancel `job_id` if it is running.
+
+        Parameters
+        ----------
+        job_id : Optional[JobId], optional
+            Job ID to cancel.  If not provided, cancel all jobs.
+        """
+
+        def end_process():
+            if os.name == 'nt':
+                self.kill()  # this might be too agressive and won't allow rollbacks!
+            else:
+                self.terminate()
+            if self._output_widget:
+                self._output_widget.append("\nTask was cancelled by the user.")
+
+        if job_id is None:
+            # cancel all jobs
+            self._queue.clear()
+            end_process()
+            return
+
+        for i, item in enumerate(self._queue):
+            if item.ident == job_id:
+                if i == 0:  # first in queue, currently running
+                    end_process()
+                else:  # still pending, just remove from queue
+                    self._queue.remove(item)
+                return
+        msg = f"No job with id {job_id}. Current queue:\n - "
+        msg += "\n - ".join(
+            [
+                f"{item.ident} -> {item.executable()} {item.arguments()}"
+                for item in self._queue
+            ]
+        )
+        raise ValueError(msg)
+
+    def waitForFinished(self, msecs: int = 10000) -> bool:
+        """Block and wait for all jobs to finish.
+
+        Parameters
+        ----------
+        msecs : int, optional
+            Time to wait, by default 10000
+        """
+        while self.hasJobs():
+            super().waitForFinished(msecs)
+        return True
+
+    def hasJobs(self) -> bool:
+        """True if there are jobs remaining in the queue."""
+        return bool(self._queue)
+
+    def set_output_widget(self, output_widget: QTextEdit):
+        if output_widget:
+            self._output_widget = output_widget
+
+    # -------------------------- Private methods ------------------------------
+    def _log(self, msg: str):
+        log.debug(msg)
+        if self._output_widget:
+            self._output_widget.append(msg)
+
+    def _build_queue_item(
+        self,
+        task_handler: TaskHandler,
+        action: ActionType,
+        pkgs: Sequence[str],
+        prefix: Optional[str] = None,
+        origins: Sequence[str] = (),
+        **kwargs,
+    ) -> AbstractInstallerTask:
+        if task_handler == "pip":
```

Comment:
We could make this more Qt like by adding an "enum" inside the class

Like


```python

class InstallerQueue(QProcess):
     
    class TaskHandler:
        conda = "conda"
        pip = "pip"
```

And then use it like:

```python
if task_handler == InstallerQueue.TaskHandler.pip:
   ...
```